### ISSUE TITLE
Quasar startup for Stack WEB3 Beginner

### ISSUE THREAD

**Issue by faithinyous** [stacksgov/Stacks-Grant-Launchpad#202]
**Background** 
Web3 D-App is still very new for most developers. Stacks has created Stack.js lib for developers to use but it could be very hard for developers to implement. The examples out there are mostly the real project and could be complex for web3 beginners. There are no starter projects that have implemented stacks.js to help developers begin their project easily. 

**Project Overview** 
In order to create the App and D-App to easily implement Web3 Technology SEEK will create the starter project using Quasar. 2 Major goals for this project are: 1 - Grow stack web3 developer which will help create and grow the community. 2 - Starter project will further develop the SEEK protocol implementation for developers to be able to use our protocol easier through stack blockchain as we already used the SEEK for sample smart contracts. 

**Scope** 
1. Create quasar project for Clarity starter 
2. Boot 
     a. Boot file for stacks.js setup 
3. Local storage *to support IOS and Android devices that have problem with store state a. UserSession 
4. Component 
     a. Connect wallet 
     b. Display user data 
     c. Token List 
          i. Token Display 
     d. Transaction History List 
          i. Transaction history 
     e. Transfer STX 
     f. Transfer Other token 
     g. Select chain 
     h. Display QR code 
5. Service 
     a. Update data - websocket 
6. Contract Owner Support Components 
     a. Mint Token 
     b. Burn Token 
7. Sample contract for testing (This SEEK token is just a template that can use some of the stack functions like mint burn and so on)
     a. SEEK-token Template with code explanation 
     b. Deploy public SEEK template on testnet 

**Budget and Milestones** 

Grant Request: 
     M1: Scope of work, deliverables, and grant amount of total. 
     Grant Request: 7000 USD 
     Estimate Duration: 25 Working days 
     Expected Complete Date: 30 Oct 
1. Create quasar project for Clarity starter 
     a. Boot file for stacks.js setup 
          i. To allow quasar to use stacks library 
     b. Local storage 
          i. to support IOS and Android devices native app that have problem with store state 
     c. UserSession plugin 
          i. Ease of access user session 
2. Component 
     a. Connect wallet 
          i. Connect to wallet in one go 
     b. Display user data 
          i. User data accessing after connect the wallet 
     c. Transfer STX 
          i. STX token transfer 
     d. Transaction History List 
          i. List of transactions in the wallet 

M2: Token transfer Component and calling smart contracts 
Grant Request: 7000 USD 
Estimate Duration: 25 Working days 
Expected Complete Date: 7 Dec 
1. Component 
     a. Token List 
          i. Display List of token in or will be in the wallet 
     b. Transfer Other token 
          i. Token in stacks chain transfer 
     c. Display QR code 
          i. QR payment
     d. Contract Owner Token management 
          i. Allow the contract owner to manage tokens such as mint, burn. 
2. Sample contract for testing (This SEEK token is just a template that can use some of the stack functions like mint burn and so on) 
     a. SEEK-token Template with code explanation 
          i. Allow other developers to launch with stacks token standard 
     b. Deploy public SEEK template on testnet 
          i. Deployment for testing 
3. Service 
     a. Live data - websocket 
          i. To display real time data to user 

**Community and Supporting Materials** 
Do you have previous projects, code commits, or experiences that are relevant to this application? What community feedback or input have you received? How do you plan to share your plan to the community over time and as the final deliverable? 

We are part of Stacks Accelerator Cohort 1, currently developing the Smart contract and backend we need to start the App and connect to our existing company.

**Comment by jennymith** [comment]
Thanks for submitting this application @faithinyous! We like the idea of a starter project that users can test but are still uncertain of exactly what you're trying to build. Can you please clarify what the end product/deliverable of this grant is?

**Comment by faithinyous** [comment]
Thank you for reviewing @jennymith. 

The product/deliverable of the grant will consist of 2 major repositories. 1st is the Vue component base that connects to stacks.js (M1.2 & M2.1). We will use a simple Vue object so that everyone can use it with the default setup component and will prepare it for allowing custom components inside the object using <slot> (not this phase). This will be pushed into npm so anybody can import and use it. 2nd repository is a quasar starter project that will be using the component from the first repository. In this part, we will implement the WebSocket as well. The last deliverable would be the smart contract that used to interact with the starter project.

In summary:
- 1st repository anyone can import from npm and use in their project. 
- 2nd repository is to learn how to use the components. 
- Smart contract to use with 2nd repository

The main reason we decided to use quasar is that the quasar framework has implemented that it will be easier to create mobile compatibility for future development for SEEKGIG.

**Comment by jennymith** [comment]
Thanks @faithinyous. We'll revisit this grant closer to January.
